src/lib/supabase/library/library.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  mockGetAllLibraryItems,
+  mockSearchLibraryItemsByAvailability,
+  mockSearchLibraryItemsByCategory,
+  mockSearchLibraryItemsByName,
+  mockSearchLibraryItemsByTags,
+  mockSearchAllLibraryItems,
+} from './mockData';
+import {
+  LibraryItem,
+  Hardware,
+  Book,
+  Film,
+  Controller,
+  Accessory,
+} from '~/models/models';
+
+//Stub Cloud Functions for the Library Page
+
+const getAllLibraryItems = (): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      resolve(mockGetAllLibraryItems);
+    }, 500);
+  });
+
+  return response;
+};
+
+const searchLibraryItemsByName = (
+  name: string
+): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      console.log('name', name);
+      resolve(mockSearchLibraryItemsByName);
+    }, 500);
+  });
+
+  return response;
+};
+
+const searchLibraryItemByTags = (
+  tags: string[]
+): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      console.log('tags', tags);
+      resolve(mockSearchLibraryItemsByTags);
+    }, 500);
```

Comment:
Curious as to why we have these `setTimeout()` methods in here.  We'll usually want to avoid anything using timeouts as it can make the code flakey.  Is it so there's enough time to get data back from the DB?  If so I'd recommend using Async/Await or `.then(resolve(mockSearchLibraryItemsByTags));` in your promise.

Reply:
sorry I was referencing this* to mock out an actual fetch. everything will be replaced with the actual async function when the back end is integrated. happy to change it to .then(resolve) but I can't exactly find documentation on the .then(resolve(mockSearchLibraryItemsByTags)). 
*https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise

Reply:
Updated in https://github.com/ws-slp/ws-slp-ts-fe/pull/40/commits/5d85d1b5056b6d6cfa6b0ab606a1d0861f169182

Reply:
There are some minor issues with this commit, left some comments!